nativelib/src/main/scala/java/lang/ClassInstancesRegistry.scala

```diff
@@ -0,0 +1,36 @@
+package java.lang
+
+/** Registry for created instances of java.lang.Class
+ * It's only purpose is to prevent GC from collecting instances of java.lang.Class
+ **/
+object ClassInstancesRegistry {
+  type Arr[T] = scala.Array[T]
```

Comment:
This type alias seems superfluous. You can directly use `Array[T]` instead. (AFAICT, the `scala.` prefix is not necessary.)